ec3730/Views/Source/SourceUrlBarView.swift

```diff
@@ -2,46 +2,83 @@ import SwiftUI
 
 struct SourceUrlBarView: View {
     @Binding var text: String
-    var refresh: () -> Void
+    var refresh: (() -> Void)?
     var go: () -> Void
+    var defaultText = "https://google.com"
+    var goText = "Go"
+    @Binding var isQuerying: Bool
+    var cancel: (() -> Void)?
 
     var body: some View {
         VStack(alignment: .leading, spacing: 0.0) {
             Divider()
+                .ignoresSafeArea()
             HStack(alignment: .center) {
-                Button {
-                    refresh()
-                } label: {
-                    Image(systemName: "arrow.clockwise")
+                if let refresh = refresh {
+                    Button {
+                        refresh()
+                    } label: {
+                        Image(systemName: "arrow.clockwise")
+                    }
                 }
-                // it would be great if this could be a .bottomBar toolbar but it is too buggy
-                TextField("https://google.com", text: $text)
-                    .textFieldStyle(RoundedBorderTextFieldStyle())
-                    .keyboardType(.URL)
-                    .disableAutocorrection(true)
 
-                Button {
-                    go()
-                } label: {
-                    Text("Go")
+                HStack {
+                    // it would be great if this could be a .bottomBar toolbar but it is too buggy
+                    TextField(defaultText, text: $text)
+                        .keyboardType(.URL)
+                        .disableAutocorrection(true)
+                        .autocapitalization(.none)
+                        .onSubmit {
+                            go()
+                        }
+
+                    if !text.isEmpty {
+                        Button(
+                            action: { text = "" },
+                            label: {
+                                Image(systemName: "delete.left")
+                                    .foregroundColor(Color(UIColor.opaqueSeparator))
+                            }
+                        )
+                    }
+                }
+                .padding(.vertical, 6)
+                .padding(.horizontal, 8)
+                .overlay(
+                    RoundedRectangle(cornerRadius: 6)
+                        .stroke(Color(uiColor: .opaqueSeparator), lineWidth: 1)
+                )
+                .background(Color(uiColor: .systemBackground).cornerRadius(6))
+
+                if isQuerying {
+                    Button {
+                        cancel?()
+                    } label: {
+                        Text("Querying")
+                    }
+                } else {
+                    Button {
```

Comment:
I think we should have some basic level of url validation and alert in this component view. And specific validation in view using it. What do you think?